### PR TITLE
Drop 0.5 support, fix some 0.7 deprecations

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

**Comment by ararslan** [comment]
Things are looking good on 0.6, but there's a pretty odd error on 0.7. There's a seemingly innocuous call:
```julia
ccall((:g_type_init, Cairo._jl_libgobject), Cvoid, ())
```
For some reason, Julia 0.7 reacts with
```
ERROR: TypeError: in init, in ccall: first argument not a pointer or valid constant expression,
expected Ptr, got Tuple{Symbol,String}
```
I'm quite certain that's unrelated to the changes in this PR, but I'm not sure what's causing it.

File: REQUIRE [hunk 1]
```diff
@@ -1,5 +1,5 @@
-julia 0.5
-Compat 0.17.0
+julia 0.6
+Compat 0.48.0
```
Comment (tkelman) [inline REQUIRE]: 0.52 for objectid https://github.com/JuliaLang/Compat.jl/commit/02367e6be37acfc319920ca593c5ae0cb9854ce4

Comment (ararslan) [reply REQUIRE]: Ah, good catch, thanks!

**Comment by bkamins** [comment]
@ararslan do you happen to know what are the plans for maintenance of this package? I personally do not use it now, but recently I got several questions if it is going to be ported to Julia 1.0 and maintained in the future.

**Comment by tknopp** [comment]
Are there depending packages? ImageView and recently also Winston have migrated to Gtk.

**Comment by bkamins** [comment]
I am not sure - but people simply want to use Tk in some legacy code I guess.

**Comment by ararslan** [comment]
> do you happen to know what are the plans for maintenance of this package?

I don't. This PR was so that Winston could work on 0.7, but as Tobias mentioned that's now using Gtk instead. The issue on 0.7 noted in this PR has been fixed in 0.7, so things should be in better shape. I don't know when I'll have time to revisit this PR since it's not exactly high priority for me, so if someone wants to start with the changes here and push it to completion then that would certainly be welcome. At this point I don't think it's worth trying to support 0.6, at least in my opinion.

**Comment by ararslan** [comment]
I reran CI on 0.7 and it appears there's still a long way to go in terms of fixing deprecations, which means 1.0 support is still a ways off.

**Pull request merged**